### PR TITLE
Run sort phase in window function execution in parallel

### PR DESCRIPTION
**Pull request by mfussenegger**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: sql/src/test/java/io/crate/execution/engine/sort/SortTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    @After
+    public void tearDownExecutor() throws Exception {
+        executor.shutdown();
+        executor.awaitTermination(5, TimeUnit.SECONDS);
+    }
+
+    @Test
+    @Repeat(iterations = 50)
```
Comment (andreidan) [inline sql/src/test/java/io/crate/execution/engine/sort/SortTest.java]: remove repeat

Comment (mfussenegger) [reply sql/src/test/java/io/crate/execution/engine/sort/SortTest.java]: This was kinda intentional given the amount of randomness within the test. It mirrors a bit property based testing where you usually get 100 runs.

Comment (andreidan) [reply sql/src/test/java/io/crate/execution/engine/sort/SortTest.java]: ah, given the various partitions that might be used... hmm, I see and agree.

**Comment by mfussenegger** [comment]
```
Q: select avg("adRevenue") over (partition by "cCode" order by "visitDate")
                from uservisits
C: 1
| Version |         Mean ±    Stdev |        Min |     Median |         Q3 |        Max |
|   V1    |     9454.300 ± 1228.576 |   8322.953 |   8779.062 |  10559.720 |  11726.378 |
|   V2    |     5556.533 ±  681.633 |   4843.495 |   5234.714 |   6250.360 |   7431.069 |
mean:   -  51.93%
median: -  50.58%
```

On a otherwise idle single node with 16 cores. With `Arrays.parallelSort` the speed up was between 80% - 90% on the same machine. I'd like to avoid using that because it doesn't support custom executors.

I guess we could look into adapting the method accordingly. Maybe something to follow up on.

**Pull request merged**